sihl/src/sihl.mli

```
@@ -1,9 +1,180 @@
+      instead. Every time you call [read_env_file] the file is read from disk. *)
+  val read_env_file : unit -> data option
+
+  (** [load_env_file ()] reads an [.env] file using {!read_env_file} and stores
+      its contents into the environment variables. *)
+  val load_env_file : unit -> unit
+
+  (** [load ()] calls {!load_env_file} and makes sure that [SIHL_ENV] was set. *)
+  val load : unit -> unit
+
+  (** [read schema] returns the decoded, statically typed version of
+      configuration [t] of the [schema]. This is used in services to
+      declaratively define a valid configuration.
+
+      The configuration data [t] is merged with the environment variable and, if
+      present, an .env file.
+
+      It fails with [Exception] and prints descriptive message of invalid
+      configuration. *)
+  val read : ('ctor, 'ty) schema -> 'ty
+
+  (** [read_string key] returns the configuration value with [key] if present.
+      The function is memoized, the first call caches the returned value and
+      subsequent calls are fast. *)
+  val read_string : string -> string option
+
+  (** [read_secret unit] returns the value of SIHL_SECRET if it is set. If
+      SIHL_SECRET was not set, it fails in production and in testing or local
+      development the value is set to "secret". *)
+  val read_secret : unit -> string
+
+  (** [read_int key] returns the configuration value with [key] if present. the
+      first call caches the returned value and subsequent calls are fast. *)
+  val read_int : string -> int option
+
+  (** [read_bool key] returns the configuration value with [key] if present. the
+      first call caches the returned value and subsequent calls are fast. *)
+  val read_bool : string -> bool option
+
+  (** [is_test ()] returns true if Sihl is running in a test environment,
+      meaning if tests are executing parts of Sihl. *)
+  val is_test : unit -> bool
+
+  (** [is_development ()] returns true if Sihl is running in a local
```

What happens if `SIHL_ENV` is not set, does it default to `development`? Could `local` be a bit misleading here?

---

Sihl forces you to set `SIHL_ENV`, I fixed the docs.

---

Thanks, but you still mention `running locally`, which could be misleading, since this function only checks the env var.

---

I adjusted the docs.

---

Thanks, looks good now :)